storagenode/pieces/readwrite.go

```diff
@@ -5,25 +5,57 @@ package pieces
 
 import (
 	"context"
+	"encoding/binary"
 	"hash"
 	"io"
 
+	"github.com/gogo/protobuf/proto"
+	"github.com/zeebo/errs"
+
+	"storj.io/storj/pkg/pb"
 	"storj.io/storj/pkg/pkcrypto"
 	"storj.io/storj/storage"
 )
 
+const (
+	// V1PieceHeaderSize is the size of the piece header used by piece storage format 1 (.sj1).
+	// It includes the size of the framing field (v1PieceHeaderFrameSize). It has a constant
+	// size because:
+	//
+	//  * we do not anticipate needing more than this
+	//  * we will be able to sum up all space used by a satellite (or all satellites) without
+	//    opening and reading from each piece file
+	//  * this simplifies piece file writing (no need to precalculate the necessary header
+	//    size before writing)
+	//
+	// If more space than this is needed, we will need to use a new storage format version.
+	V1PieceHeaderSize = 512
+
+	// v1PieceHeaderFramingSize is the size of the field used at the beginning of piece
+	// files to indicate the size of the piece header (because protobufs are not self-
+	// delimiting, which is lame).
+	v1PieceHeaderFramingSize = 2
+)
+
 // Writer implements a piece writer that writes content to blob store and calculates a hash.
 type Writer struct {
-	hash hash.Hash
-	blob storage.BlobWriter
-	size int64
+	hash     hash.Hash
+	blob     storage.BlobWriter
+	dataSize int64 // piece size only; i.e., not including piece header
```

Comment:
curious why not call this pieceSize?

Reply:
That would be fine too. I think it'd still be kind of ambiguous and would still need the clarifying comment.

Reply:
true true. yeah its tricky to know when the size contains or doesn't contain the header size